test/e2e/specs/site-editor/new-templates-list.spec.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Templates', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await Promise.all( [
+			requestUtils.activateTheme( 'emptytheme' ),
+			requestUtils.activatePlugin( 'gutenberg-test-dataviews' ),
+		] );
+	} );
+	test.afterAll( async ( { requestUtils } ) => {
+		await Promise.all( [
+			requestUtils.activateTheme( 'twentytwentyone' ),
+			requestUtils.deactivatePlugin( 'gutenberg-test-dataviews' ),
+		] );
+	} );
+	test( 'Sorting', async ( { admin, page } ) => {
+		await admin.visitSiteEditor( { path: '/wp_template/all' } );
+		// Descending by title.
+		await page.getByRole( 'button', { name: 'Template' } ).click();
+		await page.getByRole( 'menuitem', { name: 'Sort descending' } ).click();
+		const firstTitle = page
+			.getByRole( 'region', {
+				name: 'Template',
+				includeHidden: true,
+			} )
+			.getByRole( 'heading', {
```

Comment:
I think it's also beneficial to include `row` to indicate that we're selecting the title of the first row of the table. This could be rewritten into something like:

```js
const firstRowTitle = page.getByRole( 'region', { name: 'Templates' } )
  .getByRole( 'table' )
  .getByRole( 'row' )
  .first()
  .getByRole( 'heading', { level: 3 } );
```